app/src/main/java/com/example/weatherapp/models/DetailsActivity.kt

```diff
@@ -50,24 +50,19 @@ class DetailsActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?){
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_details_page)
-        initialiseComponentInfo()
-
-        val retrofithelper = RetrofitHelper.getInstance()
-        val temperatureApi = retrofithelper.create(TemperatureApi::class.java)
         supportActionBar?.hide()
 
         lifecycleScope.launch {
                 /*Main thread is responsible for rendering all the frames of the screen
                     Each piece of code originally runs on the main thread
                  */
-            val result = withContext(Dispatchers.IO){
-                temperatureApi.getTemperature(city)
+            detailsViewModel.setTemperatureDetails(city)
+            detailsViewModel.uiState.collect { uiState ->
+                setComponentInfo(uiState)
+                if (uiState.weatherInformation != null){
```

Comment:
Now that the code is complete, I think we can remove the `Log` and `comments`

Reply:
Log and comments eliminated